x/gamm/simulation/sim_msgs.go

```diff
@@ -145,7 +145,18 @@ func RandomSwapExactAmountIn(k keeper.Keeper, sim *simtypes.SimCtx, ctx sdk.Cont
 	randomCoinSubset := sim.RandSubsetCoins(sdk.NewCoins(sdk.NewCoin(accCoinIn.Denom, accCoinIn.Amount)))
 
 	// calculate the minimum number of tokens received from input of tokenIn
-	tokenOutMin, err := pool.CalcOutAmtGivenIn(ctx, randomCoinSubset, coinOut.Denom, pool.GetSpreadFactor(ctx))
+
+	// N.B. Calling MsgSwapExactAmountIn executes the swap via the pool manager, which charges the taker fee.
+	// We therefore need to remove the taker fee from the amountIn before calling the calc method.
+	takerFee, err := k.GetTradingPairTakerFee(ctx, coinIn.Denom, coinOut.Denom)
+	if err != nil {
+		return nil, err
+	}
+
+	amountInAfterSubTakerFee := randomCoinSubset[0].Amount.ToDec().Mul(sdk.OneDec().Sub(takerFee))
+	tokenInAfterSubTakerFee := sdk.NewCoin(randomCoinSubset[0].Denom, amountInAfterSubTakerFee.TruncateInt())
+
+	tokenOutMin, err := pool.CalcOutAmtGivenIn(ctx, sdk.NewCoins(tokenInAfterSubTakerFee), coinOut.Denom, pool.GetSpreadFactor(ctx))
```

Comment:
Note to reviewer: Remember, taker fees aren't charged if we call the gamm or cl method directly. This is fine though, since we always route through pool manager when submitting txs